src/uu/pidwait/src/wait.rs

```diff
@@ -3,51 +3,81 @@
 // For the full copyright and license information, please view the LICENSE
 // file that was distributed with this source code.
 
+use std::io::Result;
+use std::time::Duration;
 use uu_pgrep::process::ProcessInformation;
 
-// Dirty, but it works.
-// TODO: Use better implementation instead
-#[cfg(target_os = "linux")]
-pub(crate) fn wait(procs: &[ProcessInformation]) {
-    use std::{thread::sleep, time::Duration};
+// Reference: pidwait-any crate.
+// Thanks to @oxalica's implementation.
 
-    let mut list = procs.to_vec();
+#[cfg(target_os = "linux")]
+#[path = "./imp/linux.rs"]
+mod imp;
```

Comment:
It might make sense to add a `mod.rs` file to the `imp` folder and handle the different platforms there. It would allow you to simply use `mod imp;` here in this file, without all the `cfg` and `path` attributes. For an example, have a look at [whoami](https://github.com/uutils/coreutils/tree/main/src/uu/whoami/src) in `coreutils`.